src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs

```diff
@@ -118,16 +120,17 @@ public static object ExecuteDbDataReader(object @this, int behavior, int opCode,
             var instanceType = command.GetType();
             var commandBehavior = (CommandBehavior)behavior;
             var callingAssembly = Assembly.GetCallingAssembly();
-            var dataReaderType = callingAssembly.GetType(DbDataReader);
+            var instrumentedType = @this.GetInstrumentedType(DbCommand);
+            var dataReaderType = instrumentedType.Assembly.GetType(DbDataReader);
 
             Func<object, object, object, object> instrumentedMethod = null;
 
             try
             {
                 instrumentedMethod =
                     MethodBuilder<Func<object, object, object, object>>
-                       .Start(Assembly.GetCallingAssembly(), mdToken, opCode, nameof(ExecuteDbDataReaderAsync))
-                       .WithConcreteType(typeof(DbCommand))
+                       .Start(instrumentedType.Assembly, mdToken, opCode, nameof(ExecuteDbDataReaderAsync))
```

Comment:
These code paths look like they'll always grab the System.Data assembly but technically since ExecuteDbDataReader is protected, the method call could be made from the subclass. Are you able to test that case? Also, are you able to repro the original failing case?

Reply:
I will see if I can put something together.